code/game/turfs/simulated/floor/plating.dm

```diff
@@ -177,4 +177,44 @@
 
 /turf/simulated/floor/plating/abductor/New()
 	..()
-	icon_state = "alienpod[rand(1,9)]"
\ No newline at end of file
+	icon_state = "alienpod[rand(1,9)]"
+
+/turf/simulated/floor/plating/basalt
+	name = "basalt floor"
+	icon_state = "basalt1"
+
+/turf/simulated/floor/plating/basalt/New()
+	..()
+	icon_state = "basalt[rand(1,12)]"
+
+/turf/simulated/floor/plating/hole
+	name = "hole"
+	desc = "Yep, thats a hole."
+	icon = 'icons/turf/floors/hole.dmi'
+	icon_state = "hole"
+	smooth = SMOOTH_TRUE
+	canSmoothWith = null
+	mouse_opacity = 0
+	var/tp_x = 0 //hole drop-point
+	var/tp_y = 0
+	var/tp_z = 0
+
+/turf/simulated/floor/plating/hole/Entered(atom/movable/H)
+	if(!istype(H))
+		return
+	if(istype(H, /obj))
+		var/obj/O = H
+		O.visible_message("[O] falls down \the [src]!")
+		O.x = tp_x
+		O.y = tp_y
+		O.z = tp_z
+	else if (istype(H, /mob/living))
+		var/mob/living/M = H
+		M.visible_message("[M] falls down \the [src]!", \
+			"<span class='warning'>You fall down \the [src]!</span>")
+		M.x = tp_x
+		M.y = tp_y
+		M.z = tp_z
+		M.Weaken(6)
+		M.emote("scream")
+		M.adjustBruteLoss(20)
```

Comment:
this teleportation method is still awful, please use old systems and do not create a new one